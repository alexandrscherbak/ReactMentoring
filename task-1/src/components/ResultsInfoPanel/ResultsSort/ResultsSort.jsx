import React from 'react';
import {connect} from 'react-redux';
import SortByActions from '../../../redux/actions/sort-by.action.js';

export const ResultsSort = ({sortBy, updateSortBy}) => (
	<div>
		<span>Sort by</span>
		<button
			onClick={() => updateSortBy('release_date')}
			className={sortBy === 'release_date' ? 'active-button' : ''}
		>release date</button>
		<button
			onClick={() => updateSortBy('vote_average')}
			className={sortBy === 'vote_average' ? 'active-button' : ''}
		>rating</button>

		 <style jsx>
			{`
				button.active-button {
					background-color: pink;
				}
			`}
		</style> 
	</div>
);

export const mapStateToProps = state => {
	return {
		sortBy: state.sortBy,
	};
}

const mapDispatchToProps = dispatch => {
	return {
		updateSortBy: sortBy => dispatch(SortByActions.setSortBy(sortBy)),
	};
}

export default connect(
	mapStateToProps,
	mapDispatchToProps,
)(ResultsSort);
