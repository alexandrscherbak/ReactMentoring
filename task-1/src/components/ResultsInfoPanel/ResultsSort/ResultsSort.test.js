import React from 'react';
import Enzyme, { mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { ResultsSort, mapStateToProps } from './ResultsSort.jsx';
import renderer from 'react-test-renderer';

Enzyme.configure({ adapter: new Adapter() });

const props = {
	sortBy: 'release_date',
	updateSortBy: jest.fn(),
};

function setup() {
	return mount(<ResultsSort {...props} />);
}

describe('ResultsSort', () => {
	test('snapshot', () => {
		const component = renderer.create(
			<ResultsSort {...props}/>,
		);
		let tree = component.toJSON();
		expect(tree).toMatchSnapshot();
	});
	it('should render self and subcomponents', () => {
		const enzymeWrapper = setup();
		expect(enzymeWrapper.find('span').text()).toBe('Sort by');
		expect(enzymeWrapper.find('.active-button').text()).toBe('release date');
	});
	it('should call updateSortBy if button clicked', () => {
		const enzymeWrapper = setup();
		const button1 = enzymeWrapper.find('.active-button');
		expect(props.updateSortBy.mock.calls.length).toBe(0);
		button1.props().onClick();
		expect(props.updateSortBy.mock.calls.length).toBe(1);
		enzymeWrapper.setProps({sortBy: 'vote_average'});
		const button2 = enzymeWrapper.find('.active-button');
		button2.props().onClick();
		expect(props.updateSortBy.mock.calls.length).toBe(2);
	});
	it('should change active button if props.sortBy changed', () => {
		const enzymeWrapper = setup();
		expect(enzymeWrapper.find('.active-button').text()).toBe('release date');
		enzymeWrapper.setProps({sortBy: 'vote_average'});
		expect(enzymeWrapper.find('.active-button').text()).toBe('rating');
	});
	it('mapStateToProps should return props from state', () => {
		const state = {
			sortBy: 'sortBy',
		};
		expect(mapStateToProps(state)).toEqual({
			sortBy: 'sortBy',
		});
	});
});
