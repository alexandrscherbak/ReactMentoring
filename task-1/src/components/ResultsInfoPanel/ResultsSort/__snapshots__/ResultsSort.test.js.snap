// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResultsSort snapshot 1`] = `
<div
  className="jsx-2048650306"
>
  <span
    className="jsx-2048650306"
  >
    Sort by
  </span>
  <button
    className="jsx-2048650306 active-button"
    onClick={[Function]}
  >
    release date
  </button>
  <button
    className="jsx-2048650306 "
    onClick={[Function]}
  >
    rating
  </button>
</div>
`;
